sync-core/src/main/java/com/cloudant/sync/datastore/BasicDatastore.java

```diff
@@ -184,15 +236,12 @@ public BasicDocumentRevision getDocument(String id) {
         return getDocument(id, null);
     }
 
-    @Override
-    public BasicDocumentRevision getDocument(String id, String rev) {
-        Preconditions.checkState(this.isOpen(), "Database is closed");
-        Preconditions.checkArgument(!Strings.isNullOrEmpty(id), "DocumentRevisionTree id can not be empty");
+    private BasicDocumentRevision getDocumentInBlock(String id, String rev){
```

Comment:
I'd prefer to see methods which require you to already be in a task to take the database as an argument -- they could even be static. This makes it harder to accidentally call them from outside a submitted task.